src/main/scala/com/mwz/sonar/scala/util/syntax/SensorContext.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Sonar Scala Plugin
+ * Copyright (C) 2018 All contributors
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package com.mwz.sonar.scala
+package util
+package syntax
+
+import org.sonar.api.batch.fs.InputFile
+import org.sonar.api.batch.measure.Metric
+import org.sonar.api.batch.sensor.SensorContext
+
+import scala.language.implicitConversions
+
+object SensorContextSyntax {
+  implicit final def sensorContextSyntax(context: SensorContext): SensorContextOps =
+    new SensorContextOps(context)
+}
+
+final class SensorContextOps(val context: SensorContext) extends AnyVal {
```

Comment:
Why using an implicit conversion, instead of an implicit class?

Reply:
I've tried to achieve something similar to how the syntax package is structured in cats (see an example [here](https://github.com/typelevel/cats/blob/d117df832ab8d4a12d05e896e6a534f52e7bc6cc/core/src/main/scala/cats/synxtax/binested.scala)), but what I haven't done yet was to put everything into the package object, but I 'll do this soon ([example](https://github.com/typelevel/cats/blob/d117df832ab8d4a12d05e896e6a534f52e7bc6cc/core/src/main/scala/cats/syntax/package.scala)) - I think it's a nice way to organise implicit conversions.

> Why using an implicit conversion, instead of an implicit class?

I think that implicit classes get compiled to an implicit function and a class, it's syntax sugar, so perhaps it was done that way in cats to optimise compile times? That's the only reason I can think of.

Reply:
> "I've tried to achieve something similar to how the syntax package is structured in cats"
Oh, that would be nice - I assume we won't have an `all` shortcut.

> "I think that implicit classes get compiled to an implicit function and a class, it's syntax sugar".

I am almost sure that is correct, but it would allow us to omit the `import scala.language.implicitConversions`
Also, I have the impression that is the recommendation to use implicit classes over implicit conversions - but can't find nothing related to that, maybe I just heard about it on a conference.

> "so perhaps it was done that way in cats to optimise compile times?"

No idea...
Maybe we should ask to the very sage wizards for clarification? 🤔

Reply:
> Oh, that would be nice - I assume we won't have an `all` shortcut.

Yes, that's right - even though it's useful (and lazy!), my preference is to be more conscious about which implicit conversions I have in the scope, so I'll stay away from `all` 😆

> I am almost sure that is correct, but it would allow us to omit the import scala.language.implicitConversions
Also, I have the impression that is the recommendation to use implicit classes over implicit conversions - but can't find nothing related to that, maybe I just heard about it on a conference.

This sounds sensible, and I'm happy with making that change unless we can find any evidence strongly supporting implicit conversions. I'll also try to dig a bit deeper and see if I can find anything more about it.

Thanks for your valuable feedback! 👍

Reply:
It turns out that a value class cannot be a member of another class (trait), so if we want to have this nice structure with individual syntax traits instantiated in the package object, we need to keep the implicit conversions in those traits and have the classes sitting outside. I'll push those changes for now but feel free to propose an alternative way of doing this.

Reply:
> "It turns out that a value class cannot be a member of another class (trait)".

Oh, that explains it all.

Well, to be honest I think the only reason in **Cats** they use `traits` is to be able to combine them all in the `all` _shortcut_.
Since we don't plan to do that, we may leave them as `objects` and use _implicit classes_.

But at the end, there wouldn't be too much difference, so I will just leave it to your personal preference :+1: